rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/service/LRAService.java

```diff
@@ -410,18 +404,18 @@ void enableRecovery(@Observes @Initialized(ApplicationScoped.class) Object init)
 
     /**
      * When the deployment is unloaded unregister for recovery
-     *
-     * @param init a javax.servlet.ServletContext
      */
-    void disableRecovery(@Observes @Destroyed(ApplicationScoped.class) Object init) {
-        assert lraRecoveryModule != null;
+    @PreDestroy
+    void disableRecovery() {
+        if (lraRecoveryModule != null) {
 
-        Implementations.uninstall();
-        RecoveryManager.manager().removeModule(lraRecoveryModule, false);
-        lraRecoveryModule = null;
+            Implementations.uninstall();
+            RecoveryManager.manager().removeModule(lraRecoveryModule, false);
+            lraRecoveryModule = null;
 
-        if (LRALogger.logger.isDebugEnabled()) {
-            LRALogger.logger.debugf("LRAServicve.disableRecovery%n");
+            if (LRALogger.logger.isDebugEnabled()) {
+                LRALogger.logger.debugf("LRAServicve.disableRecovery%n");
```

Comment:
I am not sure what the %n refers to here. I think it would be fine to correct the typo given it is debug code.

Reply:
It wasn't introduced by the PR but I agree it's a good opportunity to correct it.